teuthology/repo_utils.py

```diff
@@ -255,7 +310,8 @@ def fetch_repo(url, branch, bootstrap=None, lock=True):
     src_base_path = config.src_base_path
     if not os.path.exists(src_base_path):
         os.mkdir(src_base_path)
-    dirname = '%s_%s' % (url_to_dirname(url), branch)
+    branch_dir = ref_to_dirname(branch)
+    dirname = '%s_%s' % (url_to_dirname(url), branch_dir)
```

Comment:
if `+` (concat) operator is preferred (to get as many Exceptions as possible?), then it should be used here, too

And it would be nice to have some rationale in the commit message explaining why this change (from `%s` to `+`) is being made.

Reply:
Not the case :-)